plugin/src/main/java/org/opensearch/ml/action/controller/UndeployModelControllerTransportAction.java

```diff
@@ -0,0 +1,121 @@
+
+    private final MLModelManager mlModelManager;
+    private final ClusterService clusterService;
+    private final Client client;
+    private DiscoveryNodeHelper nodeFilter;
+    private final MLStats mlStats;
+    private NamedXContentRegistry xContentRegistry;
+
+    private ModelAccessControlHelper modelAccessControlHelper;
+
+    @Inject
+    public UndeployModelControllerTransportAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        MLModelManager mlModelManager,
+        ClusterService clusterService,
+        ThreadPool threadPool,
+        Client client,
+        DiscoveryNodeHelper nodeFilter,
+        MLStats mlStats,
+        NamedXContentRegistry xContentRegistry,
+        ModelAccessControlHelper modelAccessControlHelper
+    ) {
+        super(
+            MLUndeployModelControllerAction.NAME,
+            threadPool,
+            clusterService,
+            transportService,
+            actionFilters,
+            MLUndeployModelControllerNodesRequest::new,
+            MLUndeployModelControllerNodeRequest::new,
+            ThreadPool.Names.MANAGEMENT,
+            MLUndeployModelControllerNodeResponse.class
+        );
+        this.mlModelManager = mlModelManager;
+        this.clusterService = clusterService;
+        this.client = client;
+        this.nodeFilter = nodeFilter;
+        this.mlStats = mlStats;
+        this.xContentRegistry = xContentRegistry;
+        this.modelAccessControlHelper = modelAccessControlHelper;
+    }
+
+    @Override
+    protected MLUndeployModelControllerNodesResponse newResponse(
+        MLUndeployModelControllerNodesRequest request,
+        List<MLUndeployModelControllerNodeResponse> responses,
+        List<FailedNodeException> failures
+    ) {
+        return new MLUndeployModelControllerNodesResponse(clusterService.getClusterName(), responses, failures);
+    }
+
+    @Override
+    protected MLUndeployModelControllerNodeRequest newNodeRequest(MLUndeployModelControllerNodesRequest request) {
+        return new MLUndeployModelControllerNodeRequest(request);
+    }
+
+    @Override
+    protected MLUndeployModelControllerNodeResponse newNodeResponse(StreamInput in) throws IOException {
+        return new MLUndeployModelControllerNodeResponse(in);
+    }
+
+    @Override
+    protected MLUndeployModelControllerNodeResponse nodeOperation(MLUndeployModelControllerNodeRequest request) {
+        return createUndeployModelControllerNodeResponse(request.getUndeployModelControllerNodesRequest());
+    }
+
+    private MLUndeployModelControllerNodeResponse createUndeployModelControllerNodeResponse(
+        MLUndeployModelControllerNodesRequest undeployModelControllerNodesRequest
+    ) {
+        String modelId = undeployModelControllerNodesRequest.getModelId();
+
+        Map<String, String> modelControllerUndeployStatus = new HashMap<>();
+        modelControllerUndeployStatus.put(modelId, "received");
```

Comment:
I don't see the status changed in this class. Is that by design?

Reply:
Yes, I followed other node response class. The node exception will be handled by a failure class. I used that for logging and responsing.